executor/cluster_reader.go

```diff
@@ -88,23 +95,7 @@ func (e *clusterConfigRetriever) retrieve(ctx sessionctx.Context) ([][]types.Dat
 		rows [][]types.Datum
 		err  error
 	}
-	serversInfo, err := infoschema.GetClusterServerInfo(ctx)
-	failpoint.Inject("mockClusterConfigServerInfo", func(val failpoint.Value) {
-		if s := val.(string); len(s) > 0 {
-			serversInfo = serversInfo[:0]
-			servers := strings.Split(s, ";")
-			for _, server := range servers {
-				parts := strings.Split(server, ",")
-				serversInfo = append(serversInfo, infoschema.ServerInfo{
-					ServerType: parts[0],
-					Address:    parts[1],
-					StatusAddr: parts[2],
-				})
-			}
-			// erase the error
-			err = nil
-		}
-	})
+	serversInfo, err := getClusterServerInfoWithFilter(ctx, e.extractor)
```

Comment:
It's better to pass the `set.StringSet` of `address` and `nodeTypes` to make this function reuse easily.

Reply:
great, done.